haskell/README.md

```diff
@@ -107,7 +153,13 @@ getToken = do              -- no need to manaully track the state & value of the
     _   -> modify readcahr >> pure (Token EOF)
 ```
 
-## External - Make.cmd
+## Some weird things
+
+1. How to deal with `dist-newstyle` when building with docker? Since the
+   Makefile mounts the `$pwd` to the container it generates garbage files in
+   the host filesystem. My workaround was to run `cabal clean` at the end, but
+   is there a better way?
```

Comment:
I would have liked to leave the Makefile and Dockerfile as similar to the ones from the template. I would personally do a `COPY . .` since that works with a ignore file. I think the point of that mount is to be able to run the formatter on the code. But maybe there is also some other reason? @novusnota

Reply:
So, there are multiple points of my template of a Dockerfile as I see them:
- To make it easily debuggable: there's no need to peak inside the built image for you guys, and you could simply focus on correct RUN commands
- To make it super easy for anyone not involved in a development of a particular implementation to build and see all the outputted files in the build. That makes everyone being able to mess around even if you don't know the language at all — great for learning by trial-and-error, and seeing how it works

And to deal with the junk from the position of maintainers there are three things:
- .gitignore
- `Makefile` with a `make clean` target 
- (unmerged, but was in plans) `CMD` section of Dockerfile, which would actually be `make clean && make ready`.

Reply:
Oh, and of course you can add more Dockerfiles as you see fit, say a `impl.Dockerfile`, which would suit your needs the best as a maintainer 👍